File: leet-code-solutions/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int length=INT_MAX;
        int start=0;
        int end=0;
        int current_sum=0;
        for(end=0; end<nums.size(); end++){
            current_sum +=nums[end];
            while(current_sum>=target){
                length = min(length, end-start+ 1);
                current_sum -=nums[start];
                start++;
            }
        }
        
        
        return (length !=INT_MAX) ? length: 0;
    }

};

File: DSA_second_phase/209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        i = 0
        curr = 0
        min_len = float("inf")
        for j in range(len(nums)):
            curr += nums[j]
            while curr >= target:
                min_len = min(min_len, j-i+1)
                curr -= nums[i]
                i += 1
        if sum(nums) < target:
            return 0
        else:
            return min_len